### ISSUE TITLE
Three-state compass button

### ISSUE THREAD

**Issue by thany** [organicmaps/organicmaps#3067]
MAPS.ME used to have this a long time ago, and it worked great. You would be able to tap the compass button to switch between:

* Do nothing (don't follow location or direction)
* Follow location only
* Follow location and direction

Oftenly I find myself working in the app, adding bookmarks for my next holidays and whatall, and then I don't want it to "fly" to whereever I happen to be, as soon as it has found my locations several seconds into having started or reactivated the app.

In MAPS.ME this would happen at semi-random times in the middle of doing something, and while this is definitely an issue to fix (if it hasn't been already), the threestate compass button is a perfect (to me) way to be absolutely sure the app won't fly over to my location when I don't want it to.

It automatically goes into the "do nothing" state when moving the map manually, which is good, but I still want to be able to set it "for sure" manually. And of course, setting it manually has to be persisted until changing it again manually and the setting would have to survive a restart.

**Comment by biodranik** [comment]
Isn't it already working in this way? Did you check it?

**Comment by thany** [comment]
Yes, checked, and no. It only switches between "follow location" and "follow location and direction", No amount of tapping it, will persistently disable location following.

If you think it should already work like this, please consider this a bug report, rather than a feature request.

**Comment by biodranik** [comment]
Location following should be disabled if the app can't detect current location. Any next app restart should not activate it, at least for a day (need to double-check the code for the period).

Almost everyone expects that the maps app detects the location on start. Unnecessary moving to the current position is a bug and should be fixed (e.g. if you scrolled somewhere and left the app, then returned back — you should see the same map screen, not your current position).

**Comment by thany** [comment]
This is still adding intelligence, whereas an option, like I said is just so much easier to make, and to comprehend.

Here, what's easier:
* Don't follow location if - user has panned manually, even before the app was restarted, or if location isn't immediately available.
* Don't follow location if - user has disabled it.

Option 2 is a million times easier to comprehend. I can perfectly 100% predict whether the app i going to follow my location. With option 1 (which could be the "automatic" setting) I still have to guess, and trust the app that it does what I hope it'll do.

**Comment by patepelo** [comment]
Can you please tell us if this is already fixed? I belive so

**Comment by thany** [comment]
Not on my phone. Unless, again, it has been fixed but it's not working.
So it's still a feature requests if not a bug report. Take your pick 😀
